### PR TITLE
backport of mcp251xfd (v5.10-rc3)

### PR DESCRIPTION
**Pull request by marckleinebudde**
The mcp251xfd is a CAN-FD SPI controller found on many rpi add-on modules. This pull requests ports back the driver as of v5.10-rc3.

### CODE REVIEW

File: arch/arm/boot/dts/overlays/README [hunk 1]
```diff
@@ -1762,6 +1762,21 @@ Params: oscillator              Clock frequency for the CAN controller (Hz)
 
         interrupt               GPIO for interrupt signal
 
+Name:   mcp25xxfd-spiX-Y
+Info:   Configures the MCPXXFD CAN controller family on spiX.Y
+Load:   dtoverlay=mcp25xxfd-spiX-Y,<param>=<val>
+Params: oscillator              Clock frequency for the CAN controller (Hz)
+
+        spimaxfrequency         Maximum SPI frequence (Hz)
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: There's a convention to call this property `speed`.

File: arch/arm/boot/dts/overlays/README [hunk 2]
```diff
@@ -1762,6 +1762,21 @@ Params: oscillator              Clock frequency for the CAN controller (Hz)
 
         interrupt               GPIO for interrupt signal
 
+Name:   mcp25xxfd-spiX-Y
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: Recent advances in the overlay parameter mechanism have made it possible to include the SPI bus choice as a parameter or parameters. Look at the `maxtherm` overlay to see how it's done.

File: arch/arm/boot/dts/overlays/README (hunk 1 above)
Comment (marckleinebudde) [reply arch/arm/boot/dts/overlays/README]: ok

File: arch/arm/boot/dts/overlays/README (hunk 2 above)
Comment (marckleinebudde) [reply arch/arm/boot/dts/overlays/README]: It's a pain to debug, but now it's working.

What's the best way to enable and configure the SPI chip selects for spi1 and spi2? I've found this in the README:
```
        For devices on spi1 or spi2, the interfaces should be enabled
        with one of the spi1-1/2/3cs and/or spi2-1/2/3cs overlays.
```

Comment (pelwell) [reply arch/arm/boot/dts/overlays/README]: Do what maxtherm does, i.e. assume that spi1 and spi2 have been enabled and configured as necessary using one of the spi1-* and spi2-* overlays. All this overlay has to do is disabled any spidev instance on the CS it is using.

You can push what you've got so far if you'd like me to look it over.

File: arch/arm/boot/dts/overlays/README [hunk 3]
```diff
@@ -1762,6 +1762,23 @@ Params: oscillator              Clock frequency for the CAN controller (Hz)
 
         interrupt               GPIO for interrupt signal
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README:1764]: There should be two blank lines between entries.

File: arch/arm/boot/dts/overlays/README [hunk 4]
```diff
@@ -1762,6 +1762,23 @@ Params: oscillator              Clock frequency for the CAN controller (Hz)
...
+        speed                   Maximum SPI frequence (Hz)
+
+        interrupt               GPIO for interrupt signal
+
+        rx_interrupt            GPIO for RX interrupt signal (nINT1) (optional)
+
+        xceiver_enable          GPIO for CAN transceiver enable (optional)
+
+        xceiver_active_high     specifiy if CAN transceiver enable pin is
+                                active high (optional, default: active low)
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README:1783]: And here as well.

**Review by pelwell — COMMENTED** [review]
Apart from the missing blank lines, that looks good.

File: arch/arm/boot/dts/overlays/README (hunk 3 above)
Comment (marckleinebudde) [reply arch/arm/boot/dts/overlays/README:1764]: done

File: arch/arm/boot/dts/overlays/README (hunk 4 above)
Comment (marckleinebudde) [reply arch/arm/boot/dts/overlays/README:1783]: done

**Comment by pelwell** [comment]
Thanks - I added the `spi<m>-<n>` parameter to the README and merged it offline.

**Pull request closed without merging**